File: hw/paillacho/U3/HW17Principles/MazePython/maze/Maze.py
from controller.MazeGenerator import MazeGenerator
from model.MazeModel import MazeModel
from view.MazeASCIIPrinter import MazeASCIIPrinter
# Author: Carlos Paillacho


def main():
    width = int(input("Enter the width of the maze: "))
    height = int(input("Enter the height of the maze: "))

    generator = MazeGenerator(width, height)
    grid = generator.generate()
    model = MazeModel(grid)
    printer = MazeASCIIPrinter(model)
    printer.print_maze()


if __name__ == "__main__":
    main()

File: hw/paillacho/U3/HW17Principles/MazeCode/Maze/src/main/java/ec/edu/espe/maze/Maze.java
package ec.edu.espe.maze;

import ec.edu.espe.controller.DFSMazeGenerator;
import ec.edu.espe.controller.MazeGenerator;
import ec.edu.espe.model.MazeModel;
import ec.edu.espe.view.MazeASCIIPrinter;
import ec.edu.espe.view.MazeRenderer;

import java.util.Scanner;
//Author: Carlos Paillacho
public class Maze {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter maze width: ");
        int width = scanner.nextInt();

        System.out.print("Enter maze height: ");
        int height = scanner.nextInt();

        // Usamos el generador DFS existente
        MazeGenerator generator = new DFSMazeGenerator();
        MazeModel model = generator.generate(width, height);

        // Usamos el impresor ASCII existente
        MazeRenderer printer = new MazeASCIIPrinter();
        printer.render(model);

        scanner.close();
    }
}
